CMakeLists.txt

```diff
@@ -1,7 +1,14 @@
 cmake_minimum_required(VERSION 3.5)
 project(mtk VERSION 1.0.0)
 
-add_definitions(-Wall -Wno-unused-function -march=native -msse2 -mssse3 -mfpmath=sse)
+add_definitions(-Wall -Wno-unused-function -march=native)
+include(CheckCXXCompilerFlag)
+
+check_cxx_compiler_flag("-msse2;-mssse3;-mfpmath=sse" __supports_sse3)
+if (__supports_sse3)
+    add_definitions(-msse2 -mssse3 -mfpmath=sse)
+endif()
+
```

Comment:
This whole block can be removed (except for the first line). `-march=native` includes all SSE versions that are available. Also `-mfpmath=sse` is standard on x86_64 anyways, even `-march=native` probably does not make that much difference anymore (these manual flags look like a relic from long time ago ...)

Reply:
TIL ...

Updated the PR. I added a definition to allow overriding the target arch (since we're building in the cloud for particular machines, march=native might give us some weird results)